src/internal/docker/cli.go

```diff
@@ -1,38 +1,100 @@
 package docker
 
 import (
+	"context"
 	"fmt"
+	"os"
+	"os/exec"
+	"strings"
 
+	"github.com/mitchellh/go-homedir"
 	yey "github.com/silphid/yey/src/internal"
 )
 
 type CLI struct{}
 
-func (c CLI) Start(ct yey.Context, imageTag, containerName string) error {
-	return fmt.Errorf("not implemented")
+func NewCLI() CLI {
+	return CLI{}
+}
 
-	// Get running ID and state
-	// docker ps --all --filter "name=al" --format '{{.ID}}|{{.State}}'
-	// state="exited"|"running"
+func (c CLI) Start(ctx context.Context, yeyCtx yey.Context, containerName string) error {
+	// Determine whether we need to run or exec container
+	isRunning, err := isContainerRunning(containerName)
+	if err != nil {
+		return err
+	}
+	args := []string{"run"}
```

Comment:
I think it might be clearer if instead of modified arg slices, we had smaller docker building blocks, like a `run` function and a `start` function. that way it could simply be:

```go
if running {
 return containerExec(containerName, ...)
}
return containerRun(containerName, ...)
```